File: app/src/main/java/com/example/sergejmitrofanov/librarycatalogue/interactor/BookListUseCaseImpl.java
package com.example.sergejmitrofanov.librarycatalogue.interactor;

import androidx.annotation.NonNull;
import com.example.sergejmitrofanov.librarycatalogue.entity.Book;
import java.util.List;

public class BookListUseCaseImpl implements BookListUseCase {

  @NonNull
  private final BooksSource booksSource;
  @NonNull
  private final BookListSorter bookListSorter;

  public BookListUseCaseImpl(
      @NonNull BooksSource booksSource,
      @NonNull BookListSorter bookListSorter) {
    this.booksSource = booksSource;
    this.bookListSorter = bookListSorter;
  }

  @NonNull
  @Override
  public List<Book> getBooks() {
    return bookListSorter.sortBooks(booksSource.getBooks());
  }
}

File: app/src/main/java/com/example/sergejmitrofanov/librarycatalogue/interactor/BookListUseCaseImpl.kt
package com.example.sergejmitrofanov.librarycatalogue.interactor

import javax.inject.Inject

class BookListUseCaseImpl @Inject constructor(
    private val booksSource: BooksSource,
    private val bookListSorter: BookListSorter
) : BookListUseCase {
    override fun getBooks() = bookListSorter.sortBooks(booksSource.getBooks())
}
